js/test/unit/vector/vector-tests.ts

```diff
@@ -323,3 +323,40 @@ function basicVectorTests(vector: Vector, values: any[], extras: any[]) {
         }
     });
 }
+
+// GH-45862: Make sure vectorFromArray produces the correct result for
+// FixedSizeList with null slots
+describe(`vecorFromArray() with FixedSizeList<T> and null slots`, () => {
+    test(`correct child length with null slot first`, () => {
+        let vector = vectorFromArray(
+            [null, [1, 2, 3]],
+            new FixedSizeList(3, new Field('item', new Int32())),
+        );
+        let child = vector.getChildAt(0);
+
+        expect(child).toHaveLength(6);
+        expect(child?.nullCount).toBe(3);
```

Comment:
One problem here: In Typescript, `Field()` constructor defaults to non-nullable. So the child isn't supposed to have nulls in this case.

Technically, what I was expecting is that the child would just be filled with arbitrary (zero?) values, while the outer array masked them out in the bitmap. I think if the inner array is nullable, it's a good idea to pass down the nulls. But if it's not supposed to be nullable, I think it could cause problems with IPC if we do set nulls.

Reply:
The issue this PR fixes is that the child Builder's nullBitmap isn't being expanded to represent all the slots when a null value is written into the parent Builder. So if you append 1000 null values to the `FixedSizeListBuilder`, the child's nullBitmap remain zero-sized. Since we initialize the nullBitmap with zeroes, all we need to do is grow it both when we write non-nulls as well as nulls.

In terms of Field metadata, we don't use that anywhere in JS aside from faithfully propagating it in the IPC format. We assume the contents of the nullBitmap doesn't matter to other implementations if `field.nullable == false`.

Reply:
Thanks for the clarification. [I reported this issue](https://github.com/apache/arrow/issues/45862) because I found the Arrow Rust implementation was rejecting the IPC batches being produced by this code path. I was worried that the nullability would cause incompatibilities as well, but I've just done some additional tests and it doesn't seem to be an issue. I'm good with merging these changes as is. 👍